src/dune_pkg/local_package.ml

```diff
@@ -108,16 +64,82 @@ module For_solver = struct
          (Package_dependency.list_to_opam_filtered_formula depopts)
   ;;
 
-  let opam_filtered_dependency_formula { dependencies; _ } =
-    Package_dependency.list_to_opam_filtered_formula dependencies
+  let opam_filtered_dependency_formula { dependencies; _ } = dependencies
+
+  module T = struct
+    type t = Package_name.t * OpamTypes.condition
+
+    let to_dyn = Dyn.pair Package_name.to_dyn Opam_dyn.condition
+
+    let compare (name, condition) (other_name, other_condition) =
+      let open Ordering.O in
+      let= () = Package_name.compare name other_name in
+      let compare_filters filter filter' =
+        let get_vars = function
+          | OpamTypes.Constraint _ -> []
+          | Filter filter -> Stdlib.List.sort Stdlib.compare (OpamFilter.variables filter)
```

Comment:
Why do we need to sort?

Reply:
This is basically taken taken from `vendor/opam/src/state/opamFileTools.ml` as a whole, because I wanted a way to create a `Set`/`Map` without having to implement my own `compare` function.

I assume this is because the ordering of the filters doesn't matter, if the filters are in a different ordering they should still be considered equal.

Reply:
Instead of copying this code, is it possible to just expose this code in the opamFileTools mli?

Reply:
Gone now.